build.gradle

```diff
@@ -0,0 +1,344 @@
+  from 'src/share/classes'
+  destinationDir compileJava.destinationDir
+
+  include '**/com/sun/btrace/resources/**'
+  include '**/com/sun/btrace/runtime/jaxb.index'
+  include '**/com/sun/btrace/annotations/jaxb.index'
+
+  def buildDate = (new Date()).format('YYYYMMdd')
+  filter{ it.replaceAll('\\{btrace.version\\}', "$version ($buildDate)") }
+
+  includeEmptyDirs = false
+}
+
+
+compileTestJava {
+  exclude 'traces'
+}
+
+
+processTestResources {
+  from 'src/test'
+  destinationDir compileTestJava.destinationDir
+  include 'traces/**/*.java'
+  include 'traces/**/*.xml'
+}
+
+
+task unjar(type: Copy) {
+  group 'Build'
+  description 'Explode the pre-built asm jar.'
+
+  from(zipTree("lib/btrace-asm-${asmVersion}.jar"))
+  into compileJava.destinationDir
+}
+
+
+['agent', 'boot', 'client'].each { name ->
+  tasks.create(name: "${name}Jar", type: Jar) {
+    group 'Build'
+    description "Build the btrace-${name} jar."
+    inputs.files unjar.outputs
+
+    appendix name
+    from sourceSets.main.output
+    exclude excludes["${name}"]
+    destinationDir buildDir
+    // unfortunately there's a hard-coded reference to "btrace-${name}.jar" in the code - so we have to strip the version
+    archiveName "$baseName-$appendix.$extension"
+
+    if (['agent', 'client'].contains(name)) {
+      manifest {
+        // alternatively we could "inline" the ${name}-manfifest's content
+        from file("src/share/classes/META-INF/${name}-manifest.mf")
+      }
+    }
+  }
+
+
+  tasks.create(name: "${name}SourcesJar", type: Jar) {
+    group 'Documentation'
+    description "Build the btrace-${name} sources jar."
+
+    appendix "${name}"
+    classifier 'sources'
+    from sourceSets.main.allSource
+    exclude excludes["${name}"]
+  }
+
+
+  tasks.create(name: "${name}Javadoc", type: Javadoc) {
+    group 'Documentation'
+    description "Generates Javadoc API documentation for the btrace-${name}."
+
+    title = "btrace-${name}"
+    source = sourceSets.main.allJava
+    destinationDir = reporting.file(name)
+    exclude excludes["${name}"]
+    failOnError false
+  }
+
+
+  tasks.create(name: "${name}JavadocJar", type: Jar) {
+    group 'Documentation'
+    description "Build the btrace-${name} javadoc jar."
+
+    appendix name
+    classifier 'javadoc'
+    from tasks["${name}Javadoc"].getOutputs()
+  }
+}
+
+
+task buildTraces(type: Exec) {
+  group 'Build'
+  inputs.files clientJar.outputs
+  inputs.files testClasses.outputs
+  inputs.dir "src/test/traces"
+  outputs.dir "${buildDir}/classes/test/traces"
+
+  workingDir "$projectDir"
+  environment('BTRACE_HOME', "$projectDir")
+  def btracec = (osFamily == 'windows') ? 'btracec.bat' : 'btracec'
+  executable "bin/${btracec}"
+  args "-cp ${buildDir}/classes/test"
+  args "-d ${buildDir}/classes/test"
+  args fileTree(dir: "src/test/traces", include: '**/*.java')
+
+  // It would be better gradle style to do the copying in another task . . .
+  // Do we really need this?
+  doLast {
```

Comment:
It turns out the traces source files are copied over in the processTestResources task. So, this whole block can be omitted.